efi/secureboot_policy.go

```diff
@@ -892,6 +969,14 @@ func (g *secureBootPolicyGen) processShimExecutableLaunch(branches []*secureBoot
 		flags |= shimVariableAuthorityEventsMatchSpec
 	}
 
+	version, err := shim.version()
+	if err != nil {
+		return xerrors.Errorf("cannot determine shim version: %w", err)
+	}
+	if version.Compare(shimVersion{15, 7}) >= 0 {
```

Comment:
as discussed, a gadget with a fixed shim will need to assume a snapd that stops doing this or produces branches both for the fixed shim and the buggy shim